Strategy for the volume raises

mtx250 opened the pull request:
I've been up and down these days trying the bot functions, and its amazing. I found a couple of bugs in binance exchange, and i will try to change or report them if i can't fix it. But what I wanted to talk about its that form a strategy that detects a large increase in volume. The past week , in Cryptopia, somebody buy 250.000$ in Litecoin, the nexts days the price rise 100$[any know that why raise the price, but somebody invest].  I think that any strategy for this its a mid-large range and can complemented for others type strategies. It's a happy idea I'd like to know the flaws.
Thanks for the attention.

Commit:
fix multiple trades on trend if first trade not captured from api response

talvasconcelos committed:
Strategies: Stochastic MACD (#336)

* srsi_macd strategy

* Update strategy.js

* Update strategy.js

* Update strategy.js

Commit:
Enhance backtester: support all strageties and collect vs buy hold value (#344)

* support all strageties in backtester

* collect vs buy hold value in backtester

* Update backtester.js

cmroche committed:
Add support for QuadrigaCX API (#342)

* Docker compose update to use internal data volume. Makes it work with Docker for Windows

* Initial exchange implemented

Quadriga API public and private queries are working, testing the getQuote and getBalance.

* JS Beautifier Config

* Adjusting minimum and increments for quadriga products

* Bug fixes and formatting

* Fix bug in buy function (typo) prevent callback
* Change formatting to follow the rest of zenbot

* Fix markdown percentage option description

* Fixed issues with buy command

* Fix error handling in API calls

* Prevent spamming calls to getOrder

There is a general fall through to setTimer at the end of the function that must always run, the conditionals were causing additional uncessary checks which increases probability of hitting API rate limits.

* Sell function tested and fixes

* Fix formatting remove update-selectors script

* Remove archives mapping from docker-compose.yml

Not related to this chance, didn't serve a particularly usefull purpose

* Update docker-compose.yml

kuhugo committed:
Update conf-sample.js (#313)

Added this warning to line 69 & 71. Sim seems to give non-repeatable results when using 100% balance. 

(WARNING : sim won't work properly if you set this value to 100)

kuhugo committed:
Update FAQ.md (#337)

* Update FAQ.md

Added  "How do I launch Zenbot?"
Fixed url link "limit market order"

* Update FAQ.md

cmroche committed:
Typo fix (#347)

DeviaVir committed:
Update Kraken currency pairs (#354)

META-DREAMER committed:
fix quadriga dates (#353)

Commit:
update package-lock.json

Commit:
manual mode

Commit:
interactive controls, document manual mode, fix order polling not working

Commit:
default order adjust time 30s -> 5s

Commit:
manual mode interactive toggle

Commit:
fix order cancel loop during bid/ask spread

Commit:
Typo in README.md (#358)

I think you meant S instead of C, since C is for cancelling as the line underneath explains.

mxaddict committed:
Small fix for the crash (#361)

tuxitor committed:
Bitstamp extension for Zenbot (#356)

* Bitstamp extension for Zenbot

* Synced with upstream

Commit:
fix bitstamp not found error

Commit:
Enhance backtester: support peroid, min_peroids param and rsi_macd strategy (#368)

* backtester: Support period and mini_periods

* backtester: Support rsi_macd strategy

crubb committed:
Multiple Kraken API fixes/improvements (#367)

* Kraken API:
- Fix cancelOrders (#303)
- Fix rejected trades on post only errors
- Improve detection of canceled trades
- Maybe fix sell orders executed multiple times (#339)
- Added more verbose debug logging
- Adjust timeouts for retry

* Replaced a tab with spaces

talvasconcelos committed:
Srsi macd (#369)

* updates to srsi_macd

* rename strategy to srsi_macd

* srsi_macd and sma indicator tweaks

* fixed type in length

* fixes

nedievas committed:
simple fix for correct asset display (#372)

before fix: 
zenbot balance --debug bitfinex.xrp-usd
2017-07-09 05:57:33 0.24 XRP-USD Asset: 1190.00000000 Currency: 0.75283021 Total: 285.11523020999994
after fix:
zenbot balance --debug bitfinex.xrp-usd
2017-07-09 06:08:49 0.23815000 XRP-USD Asset: 1190.00000000 Currency: 0.75283021 Total: 284.15133021

talvasconcelos committed:
Strategy: CCI + Stochasctics RSI (#375)

* updates to srsi_macd

* rename strategy to srsi_macd

* srsi_macd and sma indicator tweaks

* CCI SRSI first commit

* new strategy cci + srsi

Commit:
backtester: Rename rsi_macd to srsi_macd (#376)

nedievas committed:
fix (#373)

```node zenbot``` throws error.
Corrected.

cmroche committed:
Added cci_srsi to the backtester (#377)

crubb committed:
Only do last sell in simExit when there have been trades (#383)

cmroche committed:
Fixes for Quadriga CX trading (#386)

* Quadriga CX Fixes

* Add amount for takers fee
* Minor update to backfill warning text
* Fix variable shadowing error
* Trades with the same timestamp are now ordered oldest to newest when r eturned from getTrades
* Trade volume now returned as number and not a string, prevents NaN on volume report
* Filter results on timestamp start to prevent re-reporting the same trades in the results

* Sometime opts.from is undefined, and we should return all results.

* Formatting

dulacp committed:
setRawMode is not supported on some remote consoles (no tty) (#380)

cmroche committed:
Windows support (#387)

* Windows support

* Fixed spawn command when trading
* Fixed backtester runCommand
* Added docker-compose-windows.yml which addresses DB volume exposing issues on Windows
* Update readme.md

* Untabify file...

crubb committed:
- Bitfinex and Kraken API throw the "post only" error only on checkOrder(), check for it in engine.js (#391)

- Sometimes I would get "0% funds on hold (0.000000)". New check if balance on hold is > 0

cmroche committed:
Quadriga CX now supports LTC trading, adding to products (#392)

* Quadriga CX Fixes

* Add amount for takers fee
* Minor update to backfill warning text
* Fix variable shadowing error
* Trades with the same timestamp are now ordered oldest to newest when r eturned from getTrades
* Trade volume now returned as number and not a string, prevents NaN on volume report
* Filter results on timestamp start to prevent re-reporting the same trades in the results

* Sometime opts.from is undefined, and we should return all results.

* Formatting

* Quadriga now supports trading LTC

* Fixing QCX API handling of numbers

Some argument about the lack of type safety in javascript.

* QCX fix incorrect error reporting

natefanaro committed:
Updated name of renamed srsi_macd strategy (#400)

dhingra committed:
Removed trailing comma from dependencies. (#398)

tux-00 committed:
Add extension point in dev doc (#395)

crubb committed:
Add strategy: forex.analytics, an genetic optimization algorithm for TA-lib stats (#389)

* Add a 'noop' strategy, which just does nothing. Can be used to generate e.g. candlestick data for model training.

* - Add option to `sim` to disable output of option
- Add option to `sim` to disable writing HTML results

* Add forex_analytics training, simulation and strategy (see https://github.com/mkmarek/forex.analytics and #385)

* - .gitignore trained models
- Limit width of progressbar in training to 80 cols

* - Added timezone to datetime displays
- Fixed hang when not using a test dataset
- Changed default indicators

* Windows fixes for sim spawn

* Write a final model including sim results (output while training less pretty)

* Made train pretty again. Also write simulation Results as HTML files in models/

* Speed up parseSimulation()

* Properly name simulation result HTML files after training

* - Use the defaults of the forex.analytics project for training
- Add a check for unknown indicators

* forex_analytics documentation

* !!! -> !

tuxitor committed:
Added support for the BTCe exchange (#388)

* Added support for the BTCe exchange

* A minor cleanup

* Even more cleanup

* Visible RSI values in the free space along with the RSI graph and some minor cosmetic changes to make for a more pleasant formatting

* Adjusted display for (RSI === 100)

* Fixed a line that got lost in copy

crubb committed:
FIX: Properly check for unknown indicators in forex_analytics (#408)

benpdavison committed:
added check for message before doing anything with it (#412)

natefanaro committed:
updated c.default_selector to c.selector in README.md (#418)

Froelund committed:
Fixed docker build (#349)

* Fixed docker build

* Added --paper according to github comments

DeviaVir committed:
Strategies: TA (ema+macd) and Trust/Distrust (#285)

node-gyp and node v8.x are not friends, stick to boron for now

Rename talib's MACD (#2)

* Rename talib's MACD so it doesn't overlap with original macd

piedrogenius committed:
xmpp for trading alarms (#333)

* use of xmpp

* xmpp configs

* Update conf-sample.js

* Update engine.js

cmroche committed:
Minor fixes for Quadriga trading API (#421)

* Quadriga CX Fixes

* Add amount for takers fee
* Minor update to backfill warning text
* Fix variable shadowing error
* Trades with the same timestamp are now ordered oldest to newest when r eturned from getTrades
* Trade volume now returned as number and not a string, prevents NaN on volume report
* Filter results on timestamp start to prevent re-reporting the same trades in the results

* Sometime opts.from is undefined, and we should return all results.

* Formatting

* Quadriga now supports trading LTC

* Fixing QCX API handling of numbers

Some argument about the lack of type safety in javascript.

* QCX fix incorrect error reporting

* * Adding debug output to help troubleshoot any more issues

* Clean up use of Number(), numbro() and String() in various API calls
* Fix error with error reporting for getTrades
* Clean up error handling conditions
* Remove excess semicolons for consistency

* Fix error with retry in getBalance

* QCX - Fix look up order

* Re-enabled taker order processing in buy/sell command

This doesn't appear strictly necessary, but if it reduces API calls back to the platform it could result in fewer API over-limits.

cmroche committed:
Fix docker builds for forex.analytics (#424)

* Add missing node-gyp for forex.analytics

* Fix docker build of forex.analytics

Needs --unsafe-perm option to properly build dependant ta-lib libraries since docker run the container build with root permissions

natefanaro committed:
Support for missing win/loss and error rate from output (#426)

Froelund committed:
Improve command help for train command (#436)

Found explanations here: https://github.com/mkmarek/forex.analytics/blob/7628a98685c3b6bbedf4a4db3ddf8ae6035fdc90/include/TradingSystem.h#L33

cmroche committed:
Update exchange products for BCH support (#438)

* Updating product lists for most exchanges

Largely this is adding support for BCH
Updated: bitfinex, bittrex, kraken

Poloniex and gdax was updated, but had no changes.
BTC-e could not be updated (broken update script)
Quadriga doesn't support BCH

BCH currency pairs on Kraken don't fully match what is returned by their API. ie: API says BCH-ZUSD, but only accepts BCH-USD. This is manually updated in the product list, I suspect it is a bug that will be rectified on their side in a few days.

Tested all pairs via backfill.

* Forgot to stage the BCH fix in Kraken.

cmroche committed:
Add support for Gemini exchange platform (#445)

* Update developer document to address common bugs in getCursor

* Fix bug with implementation of getCursor

We call this function with either a trade, or timestamp in the case of a backfill

* Quadriga  - Fix bug with incorrect implementation of getCUrsor

Made some of the error reporting a bit prettier
Fix incorrect use for the numbro 'format' method

* Kraken - Fix bug with the way rejected order were handle, pass the proper status and flags now

Increase rate limit to 2000 after having some issues with backfilling

* Add support for Gemini exchange

Updated the readme to add QCX and Gemini support

* I always forget to format before submitting

* Forgot to stage conf-sample with Gemini updates

* More formatting :)

cmroche committed:
QCX added support for BCH (#452)

mesquka committed:
Added Gemini API dependancy to main package.json (#454)

DeviaVir committed:
Update package-lock.json

cmroche committed:
Fix bug with preroll when 0 volume (#464)

* Fix bug with preroll when 0 volume

Fix getCursor on Kraken exchange, getCursor must return milliseconds to be consisitent with API calls to getTrades in backfill and trade commands. It was returning seconds, this would cause entries to be returned outside of their respective time periods.

Also removed opts.to and replace with opts.from. opts.to is only used for 'backward' history scans, not forward. This was always effectively undefined in this case.

* Formatting kraken exchange.js

cmroche committed:
Introduce a non-interactive mode (#467)

* Introduce a non-interactive mode

* Fix typo

cmroche committed:
Unbreaking backfill on kraken (#468)

* Unbreaking backfill on kraken

Yeah... not sure how I managed to move that block of code.

* Since timestamps are now fixed (ms) we need to remove 3 0's from the since arg

cmroche committed:
Validated Kraken product list manually fixed some broken tokens. (#469)

crubb committed:
Bitfinex WebSockets API + Fix postonly not triggering re-buy (#458)

* Really re-order on "post only" error
Check for balance error also on checkOrder (e.g. bitfinex, using websockets, reports the error later than other exchanges)

* Initial commit to switch bitfinex to websockets

* Select pair on first getTrade()

* Fixed bitfinex buy and sell from commandline

* Bitfinex: Trigger calc of wallet balances on getBalance over websockets and retun updated balances

* Bitfinex: Removed duplicate definition of symbol

* Bitfinex: Now using only one websockets connection

* Bitfinex WS: Better error handling on connect/auth/sending messages

* Bitfinex WS: Further improving retries and error handling

* Bitfinex WS: Not directly calling the websockets part of the bitfinex module for close() anymore

* Fix bug with preroll when 0 volume (#464)

* Fix bug with preroll when 0 volume

Fix getCursor on Kraken exchange, getCursor must return milliseconds to be consisitent with API calls to getTrades in backfill and trade commands. It was returning seconds, this would cause entries to be returned outside of their respective time periods.

Also removed opts.to and replace with opts.from. opts.to is only used for 'backward' history scans, not forward. This was always effectively undefined in this case.

* Formatting kraken exchange.js

* Bitfinex WS: Fixed wallet recalc when asset or currency are 0

* Really re-order on "post only" error
Check for balance error also on checkOrder (e.g. bitfinex, using websockets, reports the error later than other exchanges)

* Initial commit to switch bitfinex to websockets

* Select pair on first getTrade()

* Fixed bitfinex buy and sell from commandline

* Bitfinex: Trigger calc of wallet balances on getBalance over websockets and retun updated balances

* Bitfinex: Removed duplicate definition of symbol

* Bitfinex: Now using only one websockets connection

* Bitfinex WS: Better error handling on connect/auth/sending messages

* Bitfinex WS: Further improving retries and error handling

* Bitfinex WS: Not directly calling the websockets part of the bitfinex module for close() anymore

* Bitfinex WS: Fixed wallet recalc when asset or currency are 0

* Bitfinex WS: New reconnect approach, honor "INSUFFICIENT MARGIN"

* Bitfinex WS: Missing )

* Bitfinex WS: Remove debug logging on "close"

cmroche committed:
Fix live trading for GDAX (and others) add a message to improve low volume pre-roll (#470)

* Show a message now while waiting for first live trade

* Fixing pre-roll trade_cursor for exchanges that don't paginate on timestamps

Commit:
Update ANS ticker to NEO (#473)

Update ANS ticker to NEO

crubb committed:
Bitfinex WS Fixes: Reconnect and crash on previously non-traded assets/currency (#474)

* Bitfinex WS: Fix reconnect

* Bitfinex WS: Report wallet balance as 0, when an unknown pair is traded for the first time on Bitfinex

* Bitfinex WS: Removed a bunch of trailing white spaces

crubb committed:
Bitfinex WS: Fix #475 wallet error (#477)

cvinacua committed:
Added Product BCH-BTC to Poloniex Exchange (#478)

DeviaVir committed:
Bittrex balances inverted. Fix xmpp. (#481)

DeviaVir committed:
Update all products.json (#482)

* Update bittrex products.json

* Update the rest as well

crubb committed:
Bitfinex WS: Attempt to fix reconnect issues on "API Key invalid" errors (#491)

* Bitfinex WS: Fix for #480?

* Bitfinex WS: Remove heartbeat setInterval on close

* Bitfinex WS: Remove redundant code

* Bitfinex WS: Fix crash on additional manual orders

* Bitfinex WS: Add missing line breaks to console.warn()/log()

nedievas committed:
ETH trading enabled (#485)

https://t.co/63rjts6NcU

cmroche committed:
FIxing kraken product update and product list (#483)

* Fixing kraken product update and product list

* Fix for Kraken's inconsistent approach to currency

Also update products as kraken recently delisted some pairs.

cvinacua committed:
Incremented decimal places for EOS asset (#510)

cvinacua committed:
Reduced min_size decimal places for EOS asset (#512)

* Incremented decimal places for EOS asset

* Reduced min_size decimal places for EOS asset

* Reduced min_size decimal places for EOS asset

nedievas committed:
Bitfinex API version update (#520)

* Bitfinex API version update

v.1.1.0

* Bitfinex API version update

v.1.1.0

nedievas committed:
Interactive CUI (#518)

Added more simple interaction options and some colors. Some early statistics on exit.

bovine committed:
fix bitstamp comment (#529)

cmroche committed:
Add a bit more data and reformat the balance command (#535)

* Add a bit more data and reformat the balance command

This allows us to see the available balance, handy when you want to make manual trades.

* Added back missing space between ask and pair

cmroche committed:
Making post_only an option (#537)

nedievas committed:
percentage format fix (#534)

is ```slippage protection: refusing to buy at 616.310 USD, slippage of +0.0%```
should be ```slippage protection: refusing to buy at 616.310 USD, slippage of +0.0015%```

Commit:
The robo kitten patch (#532)

* Update conf-sample.js

* Update conf-sample.js

nedievas committed:
percentage format fix 0.00% (#539)

nedievas committed:
percentage format fix (#540)

Wider space between high pct numbers

cmroche committed:
Package files missing 'ws' also updated (#551)

brucetus committed:
Create raspberrypi.md (#555)

BarnumD committed:
fix wording in forex analytics doc. (#556)

nedievas committed:
Interactive CUI update (#559)

Added options viewing keys

nedievas committed:
Correcting asset and currency formating (#560)

Example:
```
2017-09-24 19:03:41  200.760 ZEC-USD   -0.02%        6    -- 43    -0.0108   0.0068           0.00000727 ZEC  0.15 USD   -0.17%   +0.89%
```

nedievas committed:
npm dependencies for CUI update (#561)

* Update package-lock.json

* Update package.json

nedievas committed:
repair npm crash (#562)

benwa committed:
some missing settings (#567)

All taken from https://docs.npmjs.com/files/package.json

talvasconcelos committed:
CCI SRSI (#565)

* updates to srsi_macd

* rename strategy to srsi_macd

* srsi_macd and sma indicator tweaks

* CCI SRSI first commit

* new strategy cci + srsi

talvasconcelos committed:
updates to CCI-SRSI strategy (#568)

cmroche committed:
Refactor bootstrap to address issues with windows docker builds (#549)

* Rename zenbot.sh to zenbot.js

* Refactoring bootstrap to fix text formatting issues breaking docker on Windows

* Add back unsafe perm

* Forgot to stage the windows docker file.

* Fix duplicated line

brucetus committed:
Update README.md (#569)

evseevnn-zz committed:
Update README (#570)

More info about available strategies

cmroche committed:
Update packages-lock to unclude 'cliff' and depencies (#576)

cmroche committed:
Genetic back tester (#577)

* Adding support for a genetic backtester

* Fixing spawning with darwin script

cmroche committed:
Fix call to zenbot.sh (#579)

JensvdHeydt committed:
Filter the simulation output from ansi-codes so that the results can be properly interpreted. (#578) (#581)

The used color.js library does have a "--no-color" switch that should solve the above problem but that library isn't under active development anymore and currently has several bugs regarding the --no-color switch. Since zenbot has a very tight coupling to color.js this is the only way to solve the problem from #578.

Solves: #578

ratajczak committed:
#455 increase getTrades time window from 2 to 12 hours (#580)

DeviaVir committed:
Genetic backtester: make executable (#584)

Marbel committed:
Fix min_periods for sar strategy (#587)

Marbel committed:
Add ta_macd and ta_ema strategies for genetic backtesting (#586)

dialex committed:
remove outdated sentence (#591)

The "X days after being release" will always be outdated, so we better just remove it. It doesn't add that much info also.

nedievas committed:
CUI update (#588)

Some configurations do not display correctly trade LIVE AUTO mode.

dialex committed:
Move Discord link down + Cleanup (#592)

- Move Discord link down the documentation, so that newcomers read the quick start guide before joining the Chat and start asking questions answered by our current documentation
- Remove the words "issues" from our first heading (we don't want to scare users away)
- Review sections and headings size
- Other minor tweaks

DeviaVir committed:
Add notifiers: slack, ifttt, xmpp, pushbullet (#595)

DeviaVir committed:
Genetic algorithm to fine tune strategy parameters (#598)

JensvdHeydt committed:
Preconfigured container to use adminMongo with zenbot_mongod… (#599)

* Adding a preconfigured container to use adminMongo with zenbot_mongodb database. Visit http://<yourip>:1234 do use it

* Removed previously exposed mongodb port

JensvdHeydt committed:
Corrected Project / Connection Name for adminMongo (#602)

emabo committed:
send push messages only if in live or paper mode (#605) (#607)

this commit fixes #605

nedievas committed:
CEXIO exchange (#606)

nedievas committed:
CEXIO API upgrade (#612)

nedievas committed:
README.md info update with added exchanges (#614)

dhingra committed:
HitBTC Exchange Support (#620)

* HitBTC Exchange Support

* HitBTC Exchange Support

* HitBTC Exchange Support

Marbel committed:
fix the 50% chance that taker or maker fee getting selected (#618)

Marbel committed:
Fix kraken products for krakens update on decimal precision (#616)

DeviaVir committed:
GDAX: cancel_after (#596)

emabo committed:
change calls for zenbot script in genetic algo directory (#603)

nedievas committed:
CEXIO API not updated duo to faulty version integrity (#623)

Niller2005 committed:
Added discord notifier (#624)

* Create discord.js

* Update _codemap.js

* Update conf-sample.js

* Update conf-sample.js

DeviaVir committed:
TheRockTrading: initial setup (#622)

* TheRockTrading: initial setup

* Fix mongo error on node:latest

Niller2005 committed:
Added discord notifier (#627)

Niller2005 committed:
Fix for something i broke (#625)

* Update conf-sample.js

* Update discord.js

DeviaVir committed:
Set defaults for cancel_after and order_type (#629)

DeviaVir committed:
Cancel after fix (#631)

* Set defaults for cancel_after and order_type

* Fix cancel_after

sniper7kills committed:
Update engine.js (#632)

Added cancel_after for placeOrder

jerrytunin committed:
Pulls min_size from Bittrex instead of being a static value.  Creates a static value for min_total of 50k Satoshi (0.005BTC), Bittrex's minimum order size. (#653)

nedievas committed:
CEXIO backfill, balance and incorrect nonce bugfix (#644)

* CEXIO balance and incorrect nonce bugfix

balance undefined and nonce error crash fix. Should fix #641 issue.

* backfill fix

* quick and dirty getTrades() fix

* fix nit whitespace

brucetus committed:
Update raspberrypi.md (#651)

sniper7kills committed:
Ubuntu 16.04 Specific (#645)

* Ubuntu 16.04 Specific

Added specific step by step instructions for Ubuntu 16.04

* Update README.md

Added link to ubuntu install video

dsrikanth committed:
Adding order completion messages to notifications (#639)

emabo committed:
print stats while running trade command without exiting (#630) (#637)

Add new 'P' key for printing stats without stopping the bot

this commit fixes #630

emabo committed:
add html files for trade stats to .gitignore (#636)

nedievas committed:
HitBTC code cleaning (#633)

removing defined values but never used

FunFR committed:
Genetic backtester fix (#656)

* remove duplicate asset_capital args

* Remove sim file to reduce disk usage

* put output files in sim dir

dkarlovi committed:
docker-compose based command interface (#663)

jrots committed:
fix parsing error in genetic backtesting doesnt remove terminal colorwq (#667)

emabo committed:
Fix balance when printing stats (fix #657) (#675)

nedievas committed:
Backfill duplicated entry error fix (#672)

nedievas committed:
Fix issue #679 (#680)

emabo committed:
Add all needed options for running trust_distrust strategy on genetic_algo (#683)

bryanberger committed:
Added prowl notifier (#690)

* Added prowl notifier

* Removed unused variable: remaining. Included package-lock.json

boscoybarra committed:
Update README.md (#706)

I have updated --enable_profit_stop_pct to new option: --profit_stop_enable_pct

Note I have not changed --sell_rate=-0.006 because apart from not existing anymore in the options I don't know if this is an old reference to some other option parameter.

brucetus committed:
Update to HitBTC 2.0 (not fully functional) (#707)

* Update to HitBTC API 2.0

Buys/sells don't seem to go through, I'm still working on fixing that

* Update exchange.js

* Update update-products.sh

* Updated maker fees

Addition to the update to HitBTC 2.0

* Updated ccxt pointer

changed ccxt to latest version

KryptoNova committed:
Fix For #659 (#710)

This is a proposed fix for "Different Bittrex error #659".

KryptoNova committed:
Bittrex and Insufficient Funds (#708) (#711)

Added if statements so the exchange can have it's cut out of the purchase of the asset.
Most exchanges want their cut in BTC rather than the alt-coin being processed.
This modification allows for the exchange fee to be removed from the overall purchase price allowing for 100% of the currency balance to be spent without an insufficient funds error popping up.

KryptoNova committed:
Different Bittrex error (#659) (#715)

* Different Bittrex error (#659)

This fix should handle any error that happens and then fire's off a retry.  I tried the Try .. Catch block on fiddle and it worked flawlessly to capture the error in question.

I'm not sure how to exactly test the retry, but I created the statement similar to the one on line 78.  This technically should be recoverable, because Bittrex is just having some sort of hiccup on their API sending data.

* Needed to pass an object instead of a string.

brucetus committed:
Split markup_pct into markdown_buy_pct and markup_sell_pct (#720)

* Udate options

* update markdown options

* Split markup between buys and sells

* Split markup options between buys and sells

* Updated markup

* Updated markup for sell

KryptoNova committed:
Fix for buy_pct (#724) (#725)

* Fix for buy_pct (#724)

* Added Brackets for Readability as Requested

KryptoNova committed:
Fix poll_trades From Command Line (#730)

When using the poll_trades setting from the command line, Zenbot will ignore it in favor of the currently used conf.js file.  This effectively eliminates the possibility of an override from the command line as dictated in the settings.

This modification will allow the command line option to override the setting in the conf.js file.

KryptoNova committed:
Fix Volume in Bittrex Adapter (#729)

* Correction for Incorrect Volume in Bittrex

* Cleanup error message (#715)

Cleaned up the error message so that the console.log is gone and rely on the retry error message instead.

abduegal committed:
Update README.md (#734)

Since the code changes in https://github.com/carlos8f/zenbot/pull/720/files are not backwards compatible, I've updated the README.

mxaddict committed:
Added currency to the notifications via bots (#737)

Commit:
Threaded neural strat. This time it works :) (#718)

* reverted

* Updated Package Lock

* Fixed neural threading completely. Sort of a "Hacked Scheduler" for calculation across multiple cores. It can serve as an example for others trying to
expand the capabilities of nodejs.

* FINALLY EVERYTHING WORKS

Credits:

Brucetus for being a cool dude. Sorry I couldn't help with hitbtc for those 0.01% refunds!!!!
Harvard for the code: http://cs.stanford.edu/people/karpathy/convnetjs/docs.html
Personal inspiration.
Nodejs for including clusters in nodejs8! And symbols support with limited global variables.
Everyone else that helped me along the way threading and coding this neural strategy.
GNU Parallel for inspiration: https://www.gnu.org/software/parallel/
And all you other folks! Especially that fellow that learned me on how much processing power it takes to create a neural net.

Sincerely,
JM/TRK

* Updated, reversed signal?

For some reason this is far more profitable config

* Update strategy.js

* Updated Trendline, use markup 0.01 brother.

* Added trendline default markup

* Updated period in strategy and min_periods... Really need to promise statement preroll periods to require trade data or something.

* Updated Trend Growth to 1.001

* Added max_sell_loss_pct 0 to trendline strat

* Kind-of Sort-of fixed the preroll

* reverted to s.lookback[s.options.min_perios]

* Set growth absolutely 0 or positive to buy

* Set growth > 1 buy

* Preparing dec

* Update package.json

* Update conf-sample.js

* Update conf-sample.js

* Update strategy.js

* Update package.json

* Update strategy.js

mxaddict committed:
Updated the bitfinex products.json file (#744)

nedievas committed:
Notification update for clearer messages. (#745)

Commit:
More technical indicators + textbelt notifier (#747)

* More technical indicators and textbelt notifier

* Updated README.md and one comment

pjbollinger committed:
Fix for symmetrical option causing sim error (#751)

By specifying `Boolean` in the declaration of the option, it caused an error by defaulting to `false`. If you tried to set it to `true` using `--symmetrical=true`, this would result in an error seen on issue  #750. At this time there is still a bug with the symmetric code working.

Commit:
fix prices for bitfinex exchanger (#746)

* fix prices for bitfinex exchanger

* Update exchange.js

* Update exchange.js

* Update exchange.js

pjbollinger committed:
Fix sim symmetrical freezing (#753)

This should fix issue #752 where using the --symmetrical option in a sim causes it to freeze.

Commit:
should pass sandbox config for gemini regardless of public or private request (#754)

DeviaVir committed:
Remove BTC-E. Run update-products.sh across the exchanges. (#757)

Fixes #701

mjniuz committed:
add Bitcoin Gold in hitbtc (#759)

darland committed:
Added neural strategy for backtester (#761)

darland committed:
Fixed an issue with the exit from sim using neural strategy. (#767)

tiagosiebler committed:
Prevent poloniex returnOpenOrders API error from tricking zenbot into order completion (fixes #765). (#770)

Commit:
Wex.NZ (#771)

nedievas committed:
CEXIO error handling (#783)

* CEXIO error handling

also fixes `/home/user/zenbot/lib/engine.js:291
              order.status = api_order.status` crash.

* Update exchange.js

Commit:
Some touch-ups to wexnz exchange. (#790)

*  On branch master
 Your branch is up-to-date with 'origin/master'.

 Changes to be committed:
	modified:   conf-sample.js
	modified:   extensions/exchanges/wexnz/exchange.js
	modified:   package-lock.json
	modified:   package.json
=========================================================
package.json: bumped wexnz npm module from 1.0.2 to 1.0.3 to retain v2 api settings for now since our 1.0.2 started at v3, but work will need to be done on a new exchange.js for v3.
=========================================================
package-lock.json: routine package-lock.json update for 1.0.3 npm module.
=========================================================
conf-sample.js: added a note notifying wex.nz traders that backfilling might not support long-hauls until v2 api limits are raised from 5000 max trades or the tradeHisotry function can be fiured out..... Other exchanges just use "trades" though.
=========================================================
exchange.js: A few things, set rate limit to 1, trades count to something over ~9000 to return as many trades as the api possibly can so that the duplicate trade error does not re-occurr at smaller backfills (such as --days=0.1) and while trading at 5000 min periods.
exchange.js: Also set authed_client() typo to authedClient() as it should be.
=========================================================

Let me know if there is anything else that can be done for now, if any errors occur. I grabbed an API key, it should authenticate fine. @sinister has been authenticating, but might have bugs buying/selling at the moment.
Please reply here for any bugs, and I will try my best to iron them out in a few days.

JM ~ Sun 03 Dec 2017 05:44:08 AM CST

* Increased rate limit

* To account for BTC future prices, increased number format digits.

To account for BTC future prices, increased number format digits.

* Moved .toLowerCase up

* forgot parenthesis, moved .toLowerCase() UP

forgot parenthesis, moved .toLowerCase() UP

* Pushed empty col for better spacing.

Pushed empty col for better spacing.

tiagosiebler committed:
Add two commands to output live HTML chart without killing zenbot. Either on demand or automatically. (#785)

abduegal committed:
Added Binance to Zenbot (#782)

* Added Binance to Zenbot

* small change in the engine to make binance cancel order work

* removed the node-binance-api

* fixed feedback for pull request.

abduegal committed:
Rest API Support for zenbot (#797)

* Added Binance to Zenbot

* small change in the engine to make binance cancel order work

* removed the node-binance-api

* fixed feedback for pull request.

* Added API support to zenbot

* Added output handler in the lib

* Fixed feedback added readme

MT00x committed:
Fix kraken products decimal precision in update-products.sh (#794)

* Fix automatic products generation. Fix commi  757 that overwrote minimun increment for kraken products

* Catching more recoverable connection errors in Kraken

* Fix indenting -Issue #800 : Fix QueryOrders price

* Fix indenting

MT00x committed:
Catching more recoverable connection errors in Kraken (#805)

hvsw committed:
Corrected argument usage on example notes (#810)

Corrected argument usage on example notes

brucetus committed:
Add binance to conf.js (#819)

Everything look alright?

brucetus committed:
Stupid typo (#823)

evseevnn-zz committed:
Extended backtesting info (#832)

* Update darwin.js

Possibility set parallel process limits.
More information about Bad output.
Add selector to reports.

* fixes for darwin

fixes

Commit:
Add "engines" field to package.json (#828)

* Add "engines" field to package.json

* Add Node.js 9 to "engines" in package.json

jwalterclark committed:
Added bind_ip_all to mongodb. (#833)

Per warning:
mongodb_1       | 2017-12-11T14:05:29.348+0000 I CONTROL  [initandlisten] ** WARNING: This server is bound to localhost.
mongodb_1       | 2017-12-11T14:05:29.348+0000 I CONTROL  [initandlisten] **          Remote systems will be unable to connect to this server.
mongodb_1       | 2017-12-11T14:05:29.348+0000 I CONTROL  [initandlisten] **          Start the server with --bind_ip <address> to specify which IP
mongodb_1       | 2017-12-11T14:05:29.348+0000 I CONTROL  [initandlisten] **          addresses it should serve responses from, or with --bind_ip_all to
mongodb_1       | 2017-12-11T14:05:29.348+0000 I CONTROL  [initandlisten] **          bind to all interfaces. If this behavior is desired, start the
mongodb_1       | 2017-12-11T14:05:29.348+0000 I CONTROL  [initandlisten] **          server with --bind_ip 127.0.0.1 to disable this warning.

Resolves issue:
server_1        | warning: mongodb not accessible. some features (such as backfilling/simulation) may be disabled.
server_1        | /app/node_modules/sosa_mongo/mongo_backend.js:6
server_1        |   if (!backend_options.db) throw new Error('must pass a node-mongodb-native db with backend_options.db');
server_1        |                            ^
server_1        |
server_1        | Error: must pass a node-mongodb-native db with backend_options.db
server_1        |     at module.exports (/app/node_modules/sosa_mongo/mongo_backend.js:6:34)
server_1        |     at /app/node_modules/sosa/index.js:7:17
server_1        |     at createCollection (/app/db/createCollection.js:5:7)
server_1        |     at Object.container (/app/db/sessions.js:2:36)
server_1        |     at Object.getValue (/app/node_modules/codemap/index.js:298:22)
server_1        |     at /app/node_modules/codemap/index.js:238:23
server_1        |     at Array.forEach (<anonymous>)
server_1        |     at Object.get (/app/node_modules/codemap/index.js:237:13)
server_1        |     at get (/app/node_modules/codemap/index.js:60:22)
server_1        |     at Command.<anonymous> (/app/commands/trade.js:319:24)
server_1        |     at Command.listener (/app/node_modules/commander/index.js:315:8)
server_1        |     at Command.emit (events.js:159:13)
server_1        |     at Command.parseArgs (/app/node_modules/commander/index.js:653:12)
server_1        |     at Command.parse (/app/node_modules/commander/index.js:475:21)
server_1        |     at /app/zenbot.js:39:11
server_1        |     at /app/boot.js:28:7
server_1        |     at f (/app/node_modules/once/once.js:25:25)
server_1        |     at Glob.<anonymous> (/app/node_modules/glob/glob.js:151:7)
server_1        |     at Glob.emit (events.js:159:13)
server_1        |     at Glob._finish (/app/node_modules/glob/glob.js:197:8)
server_1        |     at done (/app/node_modules/glob/glob.js:182:14)
server_1        |     at Glob._processGlobStar2 (/app/node_modules/glob/glob.js:637:12)
zenbot_server_1 exited with code 1

MT00x committed:
Catching more recoverable connection errors in Kraken (#841)

* Fix automatic products generation. Fix commi  757 that overwrote minimun increment for kraken products

* Catching more recoverable connection errors in Kraken

* Fix indenting -Issue #800 : Fix QueryOrders price

* Fix indenting

* Catching more recoverable connection errors in Kraken

* Catching more recoverable connection errors in Kraken

* Catching generic 522 error

* Catching more recoverable connection errors in Kraken

* Catching more recoverable connection errors in Kraken

* Fix new line in debug log

MT00x committed:
Fix Kraken products minimun order size (#836)

* Fix automatic products generation. Fix commi  757 that overwrote minimun increment for kraken products

* Catching more recoverable connection errors in Kraken

* Fix indenting -Issue #800 : Fix QueryOrders price

* Fix indenting

* Catching more recoverable connection errors in Kraken

* Catching more recoverable connection errors in Kraken

* Catching generic 522 error

* Fix Kraken products minimun order size (volume) for each produtcts, according to https://support.kraken.com/hc/en-us/articles/205893708-What-is-the-minimum-order-size-

yellowcandle committed:
Added XRP and BTG selectors (#846)

MT00x committed:
Fix market order actual price (#847)

* Use actual price if possible.
In market order the actual price (api_order.price) should be very different from trade price

* Fix comment

Fix comment

darland committed:
Refactor method for notification. (#852)

tiagosiebler committed:
Add simple crossover_vwap strategy with vwap indicator. (#853)

* Add simple vwap indicator.

* Add simple vwap indicator

* Add crossover_vwap strategy, based on tests with XRP-EUR on kraken.

* Add crossover_vwap strategy, based on tests with XRP-EUR on kraken. Tweaked reporting within strat & defaults.

tiagosiebler committed:
Instead of hardcoding which values are included in the export, include all values in the export. Allows for further enhancement of the graphcs. (#854)

darland committed:
Fixed issue with send notification (#855)

tiagosiebler committed:
Add crossover_vwap strategy to readme & genetic backtester. (#858)

* Add crossover_vwap strategy to readme.

* Clean up crossover_vwap strategy source.

* Clean up phenotype create indent for int0

* Add crossover_vwap strategy to genetic backtester. Add RangeFactor function to support testing ranges at a factor of 10,100, 1000 etc.

* Add example sim results demonstrating effects of vwap_max parameter.

* Add CSS for example sim results from crossover_vwap strategy.

* Add index, for GH Pages support.

Commit:
Update strategy.js (#863)

* Update strategy.js

* Update strategy.js

* Update strategy.js

* Update strategy.js

Commit:
stddev calculate on period (#826)

* stddev calculate on period

less api calls

* Update strategy.js

* Update strategy.js

DeviaVir committed:
Create ISSUE_TEMPLATE.md

DeviaVir committed:
Create CODE_OF_CONDUCT.md (#880)

DeviaVir committed:
Create LICENSE (#881)

ehazlett committed:
Enable discord customizations (#778)

* enable discord customizations

Signed-off-by: Evan Hazlett <ejhazlett@gmail.com>

* Update discord.js

FunFR committed:
Apply sim slippage to taker order only with current price instead of orig_price (#766)

Commit:
Fixed VMA calculation (#763)

tfossoul committed:
[CHG] SIM: use human datetime format for start/end parameter (#883)

tfossoul commented:
What did u do with your PR. We shoudn't see others commits. Just yours. How could we able to know which is your to review it ?
Can you try to fix it ?

dshetyo committed:
Adding stackoverflow link to README file. (#897)

Added stackoveflow link README file. 
Link is https://stackoverflow.com/questions/tagged/zenbot

Commit:
Slow Stochastic Oscillator (#896)

defkev committed:
Improve XMPP reliability (#894)

Only connect once, keep the connection open and reconnect if necessary.

Wood-Chopper committed:
HTML report graphs use green arrows when buying (#888)

HTML report graphs use green arrows when buying

sassman committed:
make sure that a wrong API key, will yield an exception to save some time of investigating whats wrong. (#884)

can be reproduced with a correct api key but incorrect secret

tiagosiebler committed:
[Bitstamp] Fix order completion detection & balance reading during open order. (#903)

* Map order completion response from bitstamp (Finished) to one that zenbot is looking for (done). Fixes issue #901.

* • Fix currency = 0 while limit order is open at bistamp.
• Fix “sell order completed at Invalid date:” message on bitstamp order completion.
• Fix #901 preventing recognition of order completion on bitstamp.

tiagosiebler committed:
Tweak crossover_vwap genetic testing ranges/parameters for increased potential. (#905)

tiagosiebler committed:
Expansion on PR #854, missed this hardcoded section in the auto-dump (shift + D). (#907)

tiagosiebler committed:
Add exchange-specific tips to help others avoid similar headaches. (#908)

* Add exchange-specific tips to help others avoid similar headaches.

* Add comment on handling Kraken Nonce errors.

haxwell committed:
Made use of timebucket in calculating current_days_left, for consistency. (#911)

haxwell committed:
Issue #769, added code which attempts to load the configuration file passed in on the command line. If not found, tries ./conf.js. Previously, it was only trying ./conf.js (#912)

jshbrntt committed:
Fix Docker by being specific about what version of Node should be used. (#916)

Want100Cookies committed:
Add telegram notifier (#920)

* Add telegram notifier

Signed-off-by: Pascal Drewes <pascal@drewes-webdesign.nl>

* Remove confusing example chat-id

DeviaVir committed:
Add priority (#921)

DeviaVir committed:
Use numbro fork (#930)

Forked the carlos8f/numbro repo so we can merge patches.

FunFR committed:
update cexio (#931)

update fees
debug BTC-USD backfill
catch errors correctly

haxwell committed:
rename so period to so period size (#926)

* Made use of timebucket in calculating current_days_left, for consistency.

* Renamed so.period to so.periodSize. This more accurately describes what is stored in the variable. Especially because we use 'period' to describe a set of high and low values over time in engine.js. (See s.period in engine.js)

nedievas committed:
CEXIO REST version uptodate & added debug option (#936)

nedievas committed:
CEXIO api update to version 1.0.8 - solves Nonce increment error (#945)

haxwell committed:
In addition to the options on a command, there are options on (#940)

strategies. There is a callback function in engine.js which sets the
options as defined in a strategy. The strategy objects were not
updated to have a periodSize option, only a period option. So,
the periodSize attribute was not being set. This caused the
execption in this bug.

This commit adds a second option, named 'periodLength'. The name
'periodLength' was chosen, rather than the previous 'periodSize' for
consistency; the description for the --period option already said
'period length', so it made sense that the additional variable be named
that as well.

Each reference to *.periodSize has been changed to *.periodLength.

This commit continues PR #926.

Niller2005 committed:
GDAX now supports Bitcoin Cash (BCH) (#948)

Numbers taken from https://api.gdax.com/products

fuzzyTew committed:
Add asset value to balance output (#950)

fuzzyTew committed:
GDAX getQuote: return an error if market is empty (#953)

fuzzyTew committed:
Set poll time for buy/sell commands to not overflow server (#954)

nedievas committed:
CEXIO fixed wrong PR #931 for error catching (#956)

Removed annoying red retry() log. You may enable that by `--debug` option

tiagosiebler committed:
Add post-install to remove .git folder from forex.analytics npm module. Stops issue #700 from being asked about every week. (#955)

haxwell committed:
replace normalize selector paradigm with hopefully cleaner objectify selector paradigm (#906)

* Initial commit, objectify-selector.js

* Selector is now an object, and we get values from its properties, rather than being a string that we calculate as necessary to get various properties.

* Selector is now an object, and we get values from its properties, rather than being a string that we calculate as necessary to get various properties. Also, removed a do-nothing line of code.

* Removed a line of code that was normalizing the selector, but then not doing anything with it.

* Selector is now an object, and we get values from its properties, rather than being a string that we calculate as necessary to get various properties.

* Selector is now an object, and we get values from its properties, rather than being a string that we calculate as necessary to get various properties.

* Selector is now an object, and we get values from its properties, rather than being a string that we calculate as necessary to get various properties.

* Selector is now an object, and we get values from its properties, rather than being a string that we calculate as necessary to get various properties.

haxwell committed:
Added --save-dev dependencies necessary for running backtests. (#961)

haxwell committed:
Issue938 a (#960)

* In addition to the options on a command, there are options on
strategies. There is a callback function in engine.js which sets the
options as defined in a strategy. The strategy objects were not
updated to have a periodSize option, only a period option. So,
the periodSize attribute was not being set. This caused the
execption in this bug.

This commit adds a second option, named 'periodLength'. The name
'periodLength' was chosen, rather than the previous 'periodSize' for
consistency; the description for the --period option already said
'period length', so it made sense that the additional variable be named
that as well.

Each reference to *.periodSize has been changed to *.periodLength.

This commit continues PR #926.

* In addition to the options on a command, there are options on
strategies. There is a callback function in engine.js which sets the
options as defined in a strategy. The strategy objects were not
updated to have a periodSize option, only a period option. So,
the periodSize attribute was not being set. This caused the
execption in this bug.

This commit adds a second option, named 'periodLength'. The name
'periodLength' was chosen, rather than the previous 'periodSize' for
consistency; the description for the --period option already said
'period length', so it made sense that the additional variable be named
that as well.

Each reference to *.periodSize has been changed to *.periodLength.

This commit continues PR #926 and #940. Third time is the charm :)

* #926, #938, #940 - In the backtesters, renaming instances of --period to --periodLength

* #926, #938, #940 - In the backtesters, renaming instances of --period to --periodLength

* Added period and periodLength option, as they were expected for the darwin.js genetic backtest. #926 #938 #940

* #926, #938, #940 - In the (genetic) backtesters, renaming instances of --period to --periodLength

haxwell committed:
--periodLength value is kept in sim, trade, train commands (#966)

The value was being overwritten, so defaults were being applied. The
value you give on the command is now the value used, for either --period
or --periodLength.

krystophv committed:
Don't generate file if filename is none (#971)

* add .sync to gitignore

* don't generate html/tpl if filename === none

tiagosiebler committed:
Enhance MongoDB authentication logic avoid deprecated Db.prototype.authenticate call. (#972)

* Enhance MongoDB authentication logic avoid deprecated Db.prototype.authenticate call. Fixes #521

* Enhance MongoDB authentication logic avoid deprecated Db.prototype.authenticate call. Fixes #521

tiagosiebler committed:
Initialise authStr in case username is blank in conf. (#973)

tiagosiebler committed:
Add basic docs on darwin.js/ genetic backtester (#997)

* Add parent readme.

* Add gen backtesting info.

fuzzyTew committed:
Small fixes for drops in objectify-selector changes (#995)

fuzzyTew committed:
gdax error: use format described in zenbot docs (#993)

fuzzyTew committed:
GDAX: Ran update-products.sh on products.json (#991)

GDAX product details canonicalized and synchronized with API server.

leonzinger committed:
fix exchange response key (#990)

Commit:
Updated trendline "AP markup mode", profitable neural. Examples.... (#988)

* Delete strategy.js

* Add files via upload

* Update strategy.js

* Update exchange.js

* Delete conf-sample.js

* Add files via upload

* Delete strategy.js

* Add files via upload

* Update strategy.js

* Add files via upload

* Update README.md

* Update README.md

* Update README.md

* Update README.md

* Update README.md

* Update README.md

* Update README.md

* Update README.md

* Update conf-sample.js

* for 30s period

* for 30s period & arbitrage coins multi instance

* Update README.md

* Update README.md

* Update README.md

* Delete strategy.js

* Add files via upload

* Delete conf-sample.js

* Add files via upload

* Add files via upload

* Update README.md

* Add files via upload

* Update README.md

* Add files via upload

* Update conf-sample.js

* Delete Capture.PNG

* Delete Capture2.PNG

* Rename README.md to extensions/strategies/neural/README.md

* Moved for trend-line update

* Add files via upload

* Create README.md

* Update README.md

* Add files via upload

* Create README.md

* Update and rename conf-sample.js to extensions/strategies/neural/conf-example.js

* Create conf-sample.js

Commit:
Chaikin Money Flow oscillator (#987)

nedievas committed:
DEBUG option (#984)

Added live DEBUG option toggle. Press 'L' to enable/disable.
Sometimes we need to peek under the hood, when somethings pauses to work but we don't want to kill the process.

acecilia committed:
Fixed bug where the "stop loss" protection was not triggering. This was happening if the buy order was created due to a "buy stop" protection triggering. (#982)

nedievas committed:
CEXIO renew update-producs.sh (#980)

via ccxt

tiagosiebler committed:
Tweaked XRP increments for Bitstamp & fixed invalid date messages (#976)

* Tweaked XRP increments for USD & EUR on Bitstamp.

* Fix “invalid date” in order completion messages (and db tracking). This was undefined by default on bitstamp.

yang2lalang committed:
Include trendline strategy for genetic backtesting (#977)

tiagosiebler committed:
Fix typo in phenotype.js (fixes #992 & #963) (#998)

glennfu committed:
Enable support for --start and --end in darwin.js (#1002)

glennfu committed:
For #924 - Resolve bad output detected on neural strategy (#1003)

shr00mie committed:
Update darwin.js (#1006)

line period -> periodLength 541

haxwell committed:
Changed references in the Strategy objects of the backtesters to 'periodLength' rather than 'period'. (#996)

tiagosiebler committed:
Fixes for recent darwin.js errors & enhancements to per generation output (#1001)

* • best.sim was oddly undefined every generation I tried. This started in a recent PR (unsure which), so instead I’m pulling this from the first row in the ordered results. Seems to be working. This fixes #872 for now, until someone figures out where .sim was supposed to come from in the `best` object (line 623). I’ve kept the original way, in case it isn’t broken for everyone (and somehow causes issues?)
• Refactored the json & csv generation logic, so the next generation won’t start until saving finishes. Previously, darwin would misleadingly start the next generation before saving completes, often resulting in backtesting being killed too early before the files actually saved, since the assumption is if the next gen started, surely the last is finished and saved?
• Added messages when saving to file completes, including file-names (for easier resuming of generations later).
• Added additional output each generation. This includes top command from that generation, ready to copy & paste, as well as additional trade stats for those results.

* Correct typo in result output line (vsByHold->fitness). Add braces to best.sim if conditional.

tiagosiebler committed:
Revert #990. Cannot assume every currency has the same prefix, there are currencies that use the Z prefix, and #990 contradicts what list-selectors returns. (#1009)

haxwell committed:
Jasmine refactor/backfill command/creating collections service2 (#974)

* Created collection-service.js, and a unit test. Refactored backfill command to use the service.

* Added jasmine.js

DeviaVir committed:
Migrating repo

DeviaVir commented:
I unfortunately can't reopen this PR, please reopen it in case you still wish to submit this.
